Assets/Scripts/Planet.cs

```diff
@@ -7,34 +7,131 @@ public class Planet : MonoBehaviour
     public float DistanceFromShip;
     private float XDisplacement, YDisplacement;
     private Rigidbody2D PlanetRigidbody;
+    private Animator PlanetAnimator;
+    private int SetLocationPasses, PlacementPriority;
+    private const int MaxPasses = 30;
 
     private void Start()
     {
+        PlanetAnimator = GetComponent<Animator>();
+        SetLocationPasses = 0;
+        PickStartLocationRandomly();
+    }
+
+    private void Update()
+    {
+        if (PlanetAnimator.GetCurrentAnimatorStateInfo(0).IsName("PlanetDestroyed"))
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if (!collision.gameObject.CompareTag("Planet"))
+        {
+            PlanetAnimator.SetBool("PlanetHit", true);
+            Debug.Log("Game Over: A planet was destroyed by your laser.");
+        }
+        if (collision.gameObject.CompareTag("Planet"))
+        {
+            RelocateOnePlanet(collision.gameObject);
+        }
+    }
+
+    // if planets managed to get placed on top of one another twice in a row, they wouldn't reposition again without this
+    private void OnCollisionStay2D(Collision2D collision)
+    {
+        if (collision.gameObject.CompareTag("Planet"))
+        {
+            RelocateOnePlanet(collision.gameObject);
+        }
+    }
```

Comment:
We may end up turning the laser into a trigger to prevent strange collisions and pushing, which would mean that we would change these to be split into OnCollisionEnter2D and OnTriggerEnter2D (and the Stay counterparts).